### ISSUE TITLE
Screen "No data has been recorded for this website" is not always displayed

### ISSUE THREAD

**Issue by mattab** [piwik/piwik#11568]
Reproduced on demo2.piwik.org

* Create a new website called `no data`
* Go to Dashboard
* Select the `no data` website
* Expected to see https://media.githubusercontent.com/media/piwik/piwik/e2ea2a7870069e1995f301f02a99051c6de45944/tests/UI/expected-screenshots/EmptySite_emptySiteDashboard.png
* Instead got the dashboard with no data

**Comment by tsteur** [comment]
Is delete_logs_enable enabled?

**Comment by mattab** [comment]
Yes. I think in this case we should still display the helpful message when there was no data. Because if there really is no data (which could be "no data in the last 180 days", or even "no data in the last 30 days"), the screen will always be useful to be seen. But maybe we should update a little the message to clarify that

**Comment by tsteur** [comment]
Then you need to start looking into the date that is selected etc cause reports will be still available for many dates.  I don't think this has to be in 3.0.3 and don't think it is suddenly that important. It would for sure need pretty much an entirely different message when there used to be data, but is not anymore etc. 

The problem is it also gets slow as you then not only need to look in the logs but also whether there were there is a report etc. See https://github.com/piwik/piwik/pull/7365#issuecomment-84284064
